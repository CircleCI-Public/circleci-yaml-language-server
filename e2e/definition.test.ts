import {
  Commands,
  configFileUri,
  command,
  commands,
} from './utils';

describe('Definition', () => {
  it('Definition on parameter', async () => {
    const testingFile = 'config1.yml';
    const diagnostics = await commands.didOpen(testingFile);
    expect(diagnostics).toMatchSnapshot();

    const response = await command(Commands.Definition, {
      textDocument: {
        uri: configFileUri(testingFile),
      },
      position: {
        character: 16,
        line: 16,
      },
    });

    expect(response).toMatchSnapshot();
  });
});
