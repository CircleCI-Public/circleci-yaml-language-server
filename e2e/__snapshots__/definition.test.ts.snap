// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition Definition on parameter 1`] = `
DiagnosticList {
  "list": [],
}
`;

exports[`Definition Definition on parameter 2`] = `
[
  {
    "range": {
      "end": {
        "character": 28,
        "line": 481,
      },
      "start": {
        "character": 12,
        "line": 478,
      },
    },
    "uri": "file:///home/circleci/.cache/cci/orbs/.circleci/circleci/macos@2.yml",
  },
]
`;
